### ISSUE TITLE
Add API for Retrieving All Documents

### ISSUE THREAD

**Issue by devleejb** [yorkie-team/yorkie#677]
# Motivation

During the operation of our service, the need for data analysis, such as inspecting accumulated data in Yorkie Documents, has arisen.

# Description

Proposing the addition of a REST API to retrieve the contents of Yorkie Documents, enabling developers using Yorkie to utilize the data.

**Comment by krapie** [comment]
Currently, there is an `GetDocumentSummary()` admin gRPC API for getting a snapshot of a document.
Are you suggesting an HTTP REST API in JSON format to get document snapshots?

**Comment by devleejb** [comment]
Exactly. I think I need an HTTP REST API. The service data is needed more for data analysis rather than in real-time, and gRPC has significant technical overhead.

**Comment by hackerwins** [comment]
@krapie I remember connect-rpc also exposing the APIs in HTTP as well as gRPC. Am I right?
https://connectrpc.com/

**Comment by krapie** [comment]
@hackerwins Yes, connect-rpc also exposes plain HTTP along with gRPC. I think we can also resolve this issue when we switching current server rpc to connect-rpc.

**Comment by hackerwins** [comment]
Great. We will soon be using `Secret Key` for the Admin API too.
<img width="1126" alt="Screenshot 2023-11-17 at 2 25 53 PM" src="https://github.com/yorkie-team/yorkie/assets/2059311/36bfc93f-2c74-4049-9de8-21159c72c31a">

**Comment by hackerwins** [comment]
After migrating to Connect RPC, we can use REST API.

I think this issue could be solved by using the Private Key created in the Dashboard during the authentication step.

**Comment by devleejb** [comment]
Then, is it possible to provide API for other features? (Document Editing, ...)

**Comment by hackerwins** [comment]
> Then, is it possible to provide API for other features? (Document Editing, ...)

It would be better to deal with this as a separate issue.
